src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/CalculateArrayHistogram.cpp

```diff
@@ -19,27 +20,26 @@ class GenerateHistogramFromData
 {
 public:
   GenerateHistogramFromData(CalculateArrayHistogram& filter, const int32 numBins, const IDataArray& inputArray, Float64Array& histogram, std::atomic<usize>& overflow,
-                            std::tuple<bool, float64, float64>& range, size_t progIncrement)
+                            std::tuple<bool, float64, float64>& range, size_t progressIncrement)
   : m_Filter(filter)
   , m_NumBins(numBins)
   , m_InputArray(inputArray)
   , m_Histogram(histogram)
   , m_Overflow(overflow)
   , m_Range(range)
-  , m_ProgIncrement(progIncrement)
+  , m_ProgressIncrement(progressIncrement)
   {
   }
   ~GenerateHistogramFromData() = default;
 
   void operator()() const
   {
-    static_assert(std::is_base_of_v<DataObject, DataArrayType>);
-    const DataArrayType& inputArray = dynamic_cast<const DataArrayType&>(m_InputArray);
+    const auto& inputArray = dynamic_cast<const DataArray<DataArrayType>&>(m_InputArray);
     auto end = inputArray.getSize();
 
     // tuple visualization: Histogram = {(bin maximum, count), (bin maximum, count), ... }
-    float64 min = std::numeric_limits<float>::max();
-    float64 max = -1.0 * std::numeric_limits<float>::max();
+    float64 min;
+    float64 max;
```

Comment:
```suggestion
    float64 min = 0.0;
    float64 max = 0.0;
```
Even though these will get set in the following blocks, it's a good idea to avoid uninitialized variables.